### ISSUE TITLE
Can't subset a TransformedCube

### ISSUE THREAD

**Issue by felixcremer** [esa-esdl/ESDL.jl#139]
I have a cube c, which I transform to dB values via map and when I try to subset it,  I get the following error. 
When I first subset the cube and apply the dB function afterwards via map it works. 

```julia
cdB = map(dB, c)
data = subsetcube(cdB, lat=(20.691, 20.69), lon=(-98.687, -98.686))
ERROR: MethodError: no method matching subsetcube(::ESDL.Cubes.TransformedCube{Union{Missing, Float32},4,typeof(dB)}; lat=(20.691, 20.69), lon=(-98.687, -98.686))
Closest candidates are:
  subsetcube(::Dataset; var, kwargs...) at /home/qe89hep/.julia/dev/ESDL/src/Cubes/Datasets.jl:69
  subsetcube(::CubeMem{T,N} where N; kwargs...) where T at /home/qe89hep/.julia/dev/ESDL/src/Cubes/Cubes.jl:126
  subsetcube(::ESDL.Cubes.ESDLZarr.ZArrayCube{T,M,A,S} where S where A<:(Zarr.ZArray{T,N,C,S} where S<:Zarr.AbstractStore where C<:Zarr.Compressor where N) where M; kwargs...) where T at /home/qe89hep/.julia/dev/ESDL/src/Cubes/ZarrCubes.jl:367
  ...
Stacktrace:
 [1] top-level scope at none:0
```

**Comment by felixcremer** [comment]
This will be fixed by #137.

**Comment by felixcremer** [comment]
This is fixed on master
